### DISCUSSION TITLE
Image Magick Tiff Conversion errors

### DISCUSSION THREAD

**Discussion by Bharad123** [ImageMagick/ImageMagick#2250] [Help]
Hi,
Recently we updated ImageMagick version to 6.9.10.68-3.el7 from 6.7.8.9-18.el7 and this blowed up all our tiff file conversions.

Below is the error that showed up when i execute convert script.

convert: Incorrect count for "GlobalParametersIFD"; tag ignored. `TIFFFetchNormalTag' @ error/tiff.c/TIFFErrors/609.
convert: Incorrect count for "ProfileType"; tag ignored. `TIFFFetchNormalTag' @ error/tiff.c/TIFFErrors/609.
convert: no images defined `/opt/imc/dpimagingservice/p.pdf' @ error/convert.c/ConvertImageCommand/3235.

Conversion script being used:  /usr/bin/convert -compress group4 -depth 1 -units PixelsPerInch -density 300 /opt/diretory/t.tiff   /opt/directory/p.pdf

Any suggestions please.?

**Comment by fmw42** [comment]
Try reading the input first.

`/usr/bin/convert /opt/diretory/t.tiff -compress group4 -depth 1 -units PixelsPerInch -density 300 /opt/directory/p.pdf`

If that fails, post a one of your tiff files that is causing issues.

**Reply by Bharad123** [reply]
Yes. I tried reading input first.. but still same error.
Error:
convert: Unknown field with tag 444 (0x1bc) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/958.
convert: Incorrect count for "GlobalParametersIFD"; tag ignored. `TIFFFetchNormalTag' @ error/tiff.c/TIFFErrors/609.
convert: Incorrect count for "ProfileType"; tag ignored. `TIFFFetchNormalTag' @ error/tiff.c/TIFFErrors/609.
convert: no images defined `/opt/imc/dpimagingservice/p.pdf' @ error/convert.c/ConvertImageCommand/3235.

Unfortunately i had to upload my tiff file with (.pdf) extension because this browser is not allowing me to upload direct Tiff files. I just changed extension and uploaded it.


[tif.pdf](https://github.com/ImageMagick/ImageMagick/files/4892979/tif.pdf)

**Reply by fmw42** [reply]
Your file cannot be opened. Why is it a PDF suffix?  If you have trouble, zip the file, do not convert to PDF.  I would like to see your original TIFF file and not a PDF.

**Reply by Bharad123** [reply]
Attached zip.
[tif.zip](https://github.com/ImageMagick/ImageMagick/files/4893064/tif.zip)

**Reply by fmw42** [reply]
Please show the results from:

`convert -version`

and

`convert -list format | grep TIFF`

**Reply by Bharad123** [reply]
convert -version

Version: ImageMagick 6.9.10-68 Q16 x86_64 2019-11-13 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(3.1)
Delegates (built-in): bzlib cairo fontconfig freetype gslib jng jpeg lcms ltdl lzma openexr pangocairo png ps rsvg tiff wmf x xml zlib



convert -list format | grep TIFF

      EPT  EPT       rw-   Encapsulated PostScript with TIFF preview
     EPT2  EPT       rw-   Encapsulated PostScript Level II with TIFF preview
     EPT3  EPT       rw+   Encapsulated PostScript Level III with TIFF preview
   GROUP4* TIFF      rw-   Raw CCITT Group4
     PTIF* TIFF      rw+   Pyramid encoded TIFF
     TIFF* TIFF      rw+   Tagged Image File Format (LIBTIFF, Version 4.0.3)
   TIFF64* TIFF      rw+   Tagged Image File Format (64-bit) (LIBTIFF, Version 4.0.3)

**Comment by fmw42** [comment]
Your command works fine for me on IM 6.9.11.23 Q16 Mac OSX with libtiff 4.1.0.

`convert -quiet tif.tif -compress group4 -depth 1 -units PixelsPerInch -density 300 tif.pdf
`

[tif.pdf](https://github.com/ImageMagick/ImageMagick/files/4893132/tif.pdf)

I suspect either a bug in your version of ImageMagick, or a possible older version of libtif that you might be using.

**Try upgrading your libtiff**

**Reply by Bharad123** [reply]
Okay. i see my libtiff version as 4.0.3.  Let me try to update that and will give it a try again.

**Reply by fmw42** [reply]
If that fails, then try upgrading IM to its current version.

**Reply by Bharad123** [reply]
Okay. i'll try that as well. ill keep you updated. Thank you so very much for your help.
